tests/utils/verification/dockercontainer.go

```diff
@@ -0,0 +1,95 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This util provides various helper methods that can be used by different tests to
+// fetch information related to docker service and containers.
+
+package verification
+
+import (
+	"log"
+	"strings"
+
+	"strconv"
+
+	"github.com/vmware/docker-volume-vsphere/tests/constants/dockercli"
+	util "github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
+)
+
+// GetContainerName returns full container name based on given short name
+func GetContainerName(hostName, shortName string) (string, error) {
+	cmd := dockercli.ListContainers + "--filter name='" + shortName + "' --format '{{.Names}}'"
+	return ssh.InvokeCommand(hostName, cmd)
+}
+
+// GetAllContainers returns all running containers on the give host based on the filtering criteria
+func GetAllContainers(hostName, filter string) (string, error) {
+	cmd := dockercli.ListContainers + "--filter name='" + filter + "' --format '{{.Names}}'"
+	return ssh.InvokeCommand(hostName, cmd)
+}
+
+// IsDockerServiceRunning returns true if the given service is running on
+// the specified VM with specified replicas; false otherwise.
+func IsDockerServiceRunning(hostName, serviceName string, replicas int) bool {
+	log.Printf("Checking docker service [%s] status on VM [%s]\n", serviceName, hostName)
+	out, err := util.ListService(hostName, serviceName)
+	if err != nil {
+		log.Println(err)
+		return false
+	}
+
+	for i := 1; i <= replicas; i++ {
+		if !strings.Contains(out, serviceName+"."+strconv.Itoa(i)) {
+			return false
+		}
+	}
+	return true
+}
+
+// IsDockerContainerRunning returns true and the host IP if the containers are running on one of
+// the given docker hosts with specified replicas; otherwise returns false and empty string.
+func IsDockerContainerRunning(dockerHosts []string, serviceName string, replicas int) (bool, string) {
+	log.Printf("Checking running containers for docker service [%s] on docker hosts: %v\n", serviceName, dockerHosts)
+
+	//TODO: Need to implement generic polling logic for better reuse
+	for attempt := 0; attempt < 30; attempt++ {
+		status, host := isContainerRunning(dockerHosts, serviceName, replicas)
+		if status {
+			return status, host
+		}
+		misc.SleepForSec(5)
+	}
+	return false, ""
+}
+
+// isContainerRunning returns true if all replicated containers are up and running; false otherwise.
+func isContainerRunning(dockerHosts []string, serviceName string, replicas int) (bool, string) {
```

Comment:
This is a convention in Go language: unexported identifiers should start with small letter; exported identifiers should start with capital letter.